File: SurveyApplication.Domain/GuiEmail.cs
using System.ComponentModel.DataAnnotations;
using SurveyApplication.Domain.Common;

namespace SurveyApplication.Domain;

public class GuiEmail : BaseDomainEntity
{
    [Required] [MaxLength(250)] public string MaGuiEmail { get; set; }

    [Required] public string DiaChiNhan { get; set; }

    [Required] public string TieuDe { get; set; }

    [Required] public string NoiDung { get; set; }

    public int IdBangKhaoSat { get; set; }
    public int IdDonVi { get; set; }
    public int TrangThai { get; set; }
    public DateTime ThoiGian { get; set; }
}

File: SurveyApplication.UI/src/app/models/GuiEmail.ts
import { Base } from './Common/Base';
import { Paging } from './Paging';

export interface CreateGuiEmail {
  lstBangKhaoSat: number[];
  lstIdDonVi: number[];
  tieuDe: string;
  noiDung: string;
}

export interface GuiEmail extends Base {
  maGuiEmail: string;
  diaChiNhan: string;
  tieuDe: string;
  noiDung: string;
  trangThai: number;
  thoiGian: Date;
  idBangKhaoSat: number;
  tenBangKhaoSat: string;
  idDonVi: number;
  tenDonVi: string;
  nguoiThucHien: string;
  linkKhaoSat: string;
}

export interface GuiEmailBks {
  idBangKhaoSat: number;
  maBangKhaoSat: string;
  tenBangKhaoSat: string;
  countSendEmail: number;
  countSendThanhCong: number;
  countSendLoi: number;
  countSendThuHoi: number;
  ngayBatDau: Date | null;
  ngayKetThuc: Date | null;
}

export interface PagingGuiEmailBks extends Paging {
  idBanhgKhaoSat: number;
  trangThaiGuiEmail: number | null;
}

export interface GuiLaiGuiEmail {
  guiEmailDto: CreateGuiEmail;
  lstIdGuiMail: number[];
}

export interface ThuHoiGuiEmail {
  lstIdGuiMail: number[];
  diaChiNhan: string;
  tieuDe: string;
  noiDung: string;
}
